wgpu-hal/src/gles/egl.rs

```diff
@@ -532,12 +532,42 @@ impl Inner {
                 .query_string(Some(display), khronos_egl::CLIENT_APIS)
                 .unwrap()
                 .to_string_lossy();
-            client_apis
+            if client_apis
                 .split(' ')
                 .any(|client_api| client_api == "OpenGL")
+            {
+                // check if GL version is supported
+                egl.bind_api(khronos_egl::OPENGL_API).unwrap();
+                let ctx = egl
+                    .create_context(display, config, None, &[khronos_egl::NONE])
+                    .map_err(|e| {
+                        crate::InstanceError::with_source(
+                            String::from("unable to create GL context"),
+                            e,
+                        )
+                    })?;
```

Comment:
could it in theory happen that we fail creating the context while OPENGL_API is bound but wouldn't fail it if OPENGL_ES_API was bound?